AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -981,6 +981,28 @@ - (void)setContentsScaleForDisplay:(CGFloat)contentsScaleForDisplay
   _contentsScaleForDisplay = contentsScaleForDisplay;
 }
 
+- (void)applyPendingViewState
+{
+  ASDisplayNodeAssertMainThread();
+  ASDN::MutexLocker l(_propertyLock);
+
+  // FIXME: Ideally we'd call this as soon as the node receives -setNeedsLayout
```

Comment:
cc @levi, we should indeed fix this soon. I think we could probably separate out the addition of subnodes, after the node is loaded,. There is a great deal of value in ensuring the measurement pass occurs off the main thread, of course.